hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/ListingBasedRollbackStrategy.java

```diff
@@ -117,21 +126,22 @@ public List<HoodieRollbackRequest> getRollbackRequests(HoodieInstant instantToRo
               // If there is no delta commit present after the current commit (if compaction), no action, else we
               // need to make sure that a compaction commit rollback also deletes any log files written as part of the
               // succeeding deltacommit.
-              boolean higherDeltaCommits =
+              boolean hasHigherCompletedDeltaCommits =
                   !activeTimeline.getDeltaCommitTimeline().filterCompletedInstants().findInstantsAfter(commit, 1)
                       .empty();
-              if (higherDeltaCommits) {
-                // Rollback of a compaction action with no higher deltacommit means that the compaction is scheduled
+              if (hasHigherCompletedDeltaCommits && !isCommitMetadataCompleted) {
```

Comment:
How's it possible that there are higher delta commits and compaction has completed? Is restore running concurrently with ingestion?

Reply:
due to async compaction.